pkg/kubelet/cadvisor/cadvisor_linux.go

```diff
@@ -96,7 +98,7 @@ func New(imageFsInfoProvider ImageFsInfoProvider, rootPath string, cgroupRoots [
 		cadvisormetrics.AppMetrics:              struct{}{},
 		cadvisormetrics.ProcessMetrics:          struct{}{},
 	}
-	if usingLegacyStats {
+	if usingLegacyStats || utilfeature.DefaultFeatureGate.Enabled(kubefeatures.LocalStorageCapacityIsolation) {
```

Comment:
this isnt clear to me.

Reply:
this change is required for re-vendoring cAdvisor.  It won't collect any fs stats if there is no `cadvisormetrics.DiskUsageMetrics` specified.

This behaviour was changed with https://github.com/google/cadvisor/commit/e3affc1b79faa2901d7210948e92a5416166eb28

Reply:
this feature is on by default... this effectively makes the default cadvisor config always include disk usage metrics. is that expected?

Reply:
without this change, cAdvisor won't collect any fs stats: https://github.com/google/cadvisor/commit/e3affc1b79faa2901d7210948e92a5416166eb28#diff-061d1d5fc291bd4b18893f2ea01457ccR191-R194

That is expected given the change of behaviour in cAdvisor but it would be probably better to fix it in cAdvisor and have a way to retrieve only the global fs info.